src/App.jsx

```diff
@@ -14,45 +16,120 @@ export const goodsFromServer = [
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_NAME = 'Name';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPrepearedGoods(goods, { currentSortField, isReversed }) {
+  const prepearedGoods = [...goods];
+
+  if (currentSortField !== '') {
+    prepearedGoods.sort((good1, good2) => {
+      switch (currentSortField) {
+        case SORT_FIELD_NAME: {
+          return good1.localeCompare(good2);
+        }
+
+        case SORT_FIELD_LENGTH: {
+          return good1.length - good2.length;
+        }
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return isReversed ? prepearedGoods.reverse() : prepearedGoods;
+}
+
+export const App = () => {
+  const [currentSortField, setCurrentSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const vissibleGoods = getPrepearedGoods(goodsFromServer,
+    { currentSortField, isReversed });
+
+  const isStateNotReversedOrSorted
+    = isReversed || currentSortField;
+
+  const sortByName = () => {
+    setCurrentSortField(SORT_FIELD_NAME);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const sortByLength = () => {
+    setCurrentSortField(SORT_FIELD_LENGTH);
+  };
+
+  const reverse = () => {
+    if (isReversed) {
+      setIsReversed(false);
+    }
+
+    if (!isReversed) {
+      setIsReversed(true);
+    }
+  };
+
+  const reset = () => {
+    setIsReversed(false);
+    setCurrentSortField('');
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={sortByName}
+          type="button"
+          className={classnames(
+            'button',
+            'is-info',
+            { 'is-light': currentSortField !== SORT_FIELD_NAME },
+          )
+          }
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={sortByLength}
+          type="button"
+          className={classnames(
+            'button',
+            'is-success',
+            { 'is-light': currentSortField !== SORT_FIELD_LENGTH },
+          )
+          }
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={reverse}
+          type="button"
+          className={
+            classnames('button is-warning',
+              { 'is-light': !isReversed })
+          }
```

Comment:
The same fix is needed as you did in other classanames, pay attention to fix in all places